pkg/kubelet/eviction/eviction_manager.go

```diff
@@ -496,23 +502,56 @@ func (m *managerImpl) emptyDirLimitEviction(podStats statsapi.PodStats, pod *v1.
 			}
 		}
 	}
+
+	return false
+}
+
+func (m *managerImpl) podEphemeralStorageLimitEviction(podStats statsapi.PodStats, pod *v1.Pod) bool {
+	_, podLimits := apiv1resource.PodRequestsAndLimits(pod)
+	_, found := podLimits[v1.ResourceEphemeralStorage]
+	if !found {
+		return false
+	}
+
+	podEphemeralStorageTotalUsage := &resource.Quantity{}
+	fsStatsSet := []fsStatsType{}
+	if *m.dedicatedImageFs {
```

Comment:
sorry if this is already discussed somewhere; i have two questions regarding dedicated image fs:
- why do we still count `fsStatsLogs` if we have dedicated imagefs?  IIUC, it is yet to decide how to handle container logs, it's possible that imagefs provides storage for logs?
- are we simply ignoring eviction for imagefs? e.g. if i have imagefs setup and a container keeps writing to writable layer, it can use up all storage in imagefs?

Reply:
We only manage one partition(root) now.  If we do not have imagefs partition, then writable layer is satisfied by root partition, we need to calculate its usage, if we have imagefs partition,  do not manage it now.

Reply:
which means this is true? `if i have imagefs setup and a container keeps writing to writable layer, it can use up all storage in imagefs?` 
> We only manage one partition(root) now. If we do not have imagefs partition, then writable layer is satisfied by root partition, we need to calculate its usage, if we have imagefs partition, do not manage it now.

Reply:
@ddysher yes, that is true..